### PR TITLE
CI for main

### PR DESCRIPTION
**Pull request by sjalander**

### CODE REVIEW

File: .github/actions/InstallPackages/action.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        fi
+      shell: bash
+
+    - name: "Update apt sources"
+      run: sudo apt-get update
+      shell: bash
+
+    - name: "Install LLVM and clang package"
+      run: |
+        sudo apt-get install llvm-${{inputs.llvm-version}}-dev clang-${{inputs.llvm-version}} mlir-${{inputs.llvm-version}}-tools libmlir-${{inputs.llvm-version}}-dev
```
Comment (phate) [inline .github/actions/InstallPackages/action.yml]: The `ninja-build` and `verilator` package can be installed together with the other packages.

Comment (phate) [reply .github/actions/InstallPackages/action.yml]: Not a big thing though. It is not much overhead of installing them individually. However, I do not see really the need to install them individually if the option of excluding them from the installation, like in the jlm CI, is not really provided any longer.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        make -j `nproc` -O all
+      shell: bash
+
+    - name: "MLIR libs"
+      run: |
+        if ! [ -f /usr/lib/x86_64-linux-gnu/libMLIR.so ]; then
+          sudo ln -s /usr/lib/llvm-17/lib/libMLIR.so.17 /usr/lib/x86_64-linux-gnu/
+          sudo ln -s /usr/lib/llvm-17/lib/libMLIR.so.17 /usr/lib/x86_64-linux-gnu/libMLIR.so
+        fi
+      shell: bash
```
Comment (phate) [inline .github/workflows/tests.yml]: Could we introduce a environment variable `LLMV_VERSION` where we set the LLVM version and then use it here as well as in the InstallPackages script? In other words, one place to rule it all such that we only have to change this one place when we update LLVM in the future.

Comment (sjalander) [reply .github/workflows/tests.yml]: These shouldn't have to be here and are annoying. It runs on my machine locally, but for some reason the LD_LIBRARY_PATH in the makefile gets ignored in the CI.

File: .github/actions/InstallPackages/action.yml (hunk 1 above)
Comment (sjalander) [reply .github/actions/InstallPackages/action.yml]: All combined to a single install

File: .github/workflows/tests.yml (hunk 2 above)
Comment (sjalander) [reply .github/workflows/tests.yml]: Added 17 as LLVM_VERSION variable

File: .github/actions/InstallPackages/action.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+name: "Install packages"
+description: "Installs packages that the hls-test-suite depends on."
+
+inputs:
+  llvm-version:
+    description: "LLVM/ version that is installed"
+    default: "17"
+    required: false
```
Comment (phate) [inline .github/actions/InstallPackages/action.yml]: Remove the `default` and make the required `true`, so it is necessary to hand it in from the site where it is invoked.

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+  hls-test-suite:
+    runs-on: ubuntu-22.04
+    if: contains(github.event.pull_request.title, '[DisableCI]') == false
+    steps:
+    - uses: actions/checkout@v4
+    - name: "Install packages"
+      uses: ./.github/actions/InstallPackages
+      with:
+        llvm-version: 17
```
Comment (phate) [inline .github/workflows/tests.yml:20]: I guess it is not possible to use the environment variable here, right? Is there a way to use something similar like `inputs.llvm-version` here like in the composite actions?

It seems there exist something similar that you can use: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-of-onworkflow_callinputs

**Review by phate — CHANGES_REQUESTED** [review]
I am sorry, but I will send this one more time back. You did not see the LLVM version through. There are still two places where this needs to be specified. I think it is possible to have only one place where to specify this.

File: .github/actions/InstallPackages/action.yml (hunk 3 above)
Comment (sjalander) [reply .github/actions/InstallPackages/action.yml]: Removed default and changed to required.

File: .github/workflows/tests.yml (hunk 4 above)
Comment (sjalander) [reply .github/workflows/tests.yml:20]: I didn't get the inputs to work. The documentation is for callable workflows, so maybe it only works when called from a different workflow.
I decided to move the MLIR symlinks to the InstallPackage action, which results in the LLVM version only having to be specified once.

**Review by phate — APPROVED** [review]

**Pull request merged**